queries/course_requests.js

```diff
@@ -1,31 +1,141 @@
+                if (err) {
+                    console.log(err);
+                    res.status(400).json({
+                        error: err
+                    });
+                } else {
+
+                    res.status(200).json({
+                        "result": formatResult(recordset.recordset),
+                    });
+
+                    updateCourseRequests(sql);
+                }
+            });
+    },
+
+    /**
+     * Get the total number of course requests in the database.
+     *
+     * A status of 200 is given for successful requests. A status of 400 for unsuccessful ones.
+     *
+     * @param {*} sql The mssql instance connected to the database currently used by the API.
+     * @param {*} res The object to send result to a given query sender.
+     */
+    getNumberOfCourseRequests: function (sql, res) {
+        const request = new sql.Request();
+
+        request.query(
+            "SELECT COUNT(ID) FROM CourseRequests WHERE Status=0",
+            function (err, recordset) {
+                if (err) {
+                    res.status(400).json({
+                        error: err
+                    });
+                }
+
+                res
+                    .status(200)
+                    .json({
+                        number: Object.values(recordset.recordset[0])[0]
+                    });
+            }
+        );
+    },
+
+}
+
+/**
+ * Updates the status of the new course requests to pending (= 1).
+ * 
+ * @param {*} sql The mssql instance connected to the database currently used by the API.
+ */
+function updateCourseRequests(sql) {
+
     const request = new sql.Request();
 
-    request.query(
-      "SELECT COUNT(ID) FROM CourseRequests WHERE Status=0",
-      function (err, recordset) {
-        if (err) {
-          res.status(400).json({ error: err });
+    request
+        .query('update CourseRequests set status = 1 where status = 0', function (err) {
+            if (err) {
+                console.log(err);
+            } else {
+                console.log("New course request(s) have been updated.");
+            }
+        });
+}
+
+/**
+ * The method allows to get a well formatted result for the new course request route.
+ * Each course request record now has a property dateOptions that gathers all the date options in a list. 
+ * 
+ * @param {*} result This is the list of all the new course requests.
+ * @returns The arranged list of all the new course requests having a property date options.
+ */
+function formatResult(result) {
+
+    let listOfSubjects = [];
+    let resultWanted = [];
+
+    for (const newCourseRequest of result) {
+        if (!listOfSubjects.includes(newCourseRequest.Subject)) {
+            listOfSubjects.push(newCourseRequest.Subject);
         }
+    }
+
+    for (const subject of listOfSubjects) {
+        let informationToRetrieve = {
```

Comment:
I think u don't have to do that.
You can simply:
informationToRetrive = {"ID":courseRequest.ID}, "Subject":courseRequest.Subject,...} on line 119 and don't need to include all those undefined lines of definition

Reply:
u right